openhtf/io/output/console_summary.py

```diff
@@ -0,0 +1,113 @@
+"""Module to display test summary on console."""
+import os
+import sys
+
+from openhtf import util
+from openhtf.io import test_record
+from openhtf.util import measurements as measurement
+
+
+class ConsoleSummary():
+  """Print test results with failure info on console. """
+
+  # pylint: disable=invalid-name
+  def __init__(self):
+    if os.name == 'posix':    #Linux and Mac
+      self.RED = '\033[91m'
+      self.GREEN = '\033[92m'
+      self.ORANGE = '\033[93m'
+      self.RESET = '\033[0m'
+      self.BOLD = '\033[1m'
+    else:
+      self.RED = ""
+      self.GREEN = ""
+      self.ORANGE = ""
+      self.RESET = ""
+      self.BOLD = ""
+
+    self.color_table = {
+        test_record.Outcome.PASS:self.GREEN,
+        test_record.Outcome.FAIL:self.RED,
+        test_record.Outcome.ERROR:self.ORANGE,
+        test_record.Outcome.TIMEOUT:self.ORANGE,
+    }
+  # pylint: enable=invalid-name
+
+  def __call__(self, record):
+    indent2 = ''.join(('\n', ' '*2))
+    indent4 = ''.join(('\n', ' '*4))
+
+    # get status string from outcome enum
+    status = str(record.outcome)[-5:]
+    if status[0] == '.':
+      status = status[1:]
+
+    color = self.color_table[record.outcome]
+
+    output_lines = []
+    msg = ''.join((color,
+                   self.BOLD,
+                   os.path.basename(sys.argv[0]),
+                   ': ',
+                   status,
+                   self.RESET,
+                   '\n'))
+
+    output_lines.append(msg)
+
+    if record.outcome in (test_record.Outcome.FAIL, test_record.Outcome.ERROR):
+      phases = record.phases
+      for phase in phases:
+        phase_name = phase.name
+        phase_time = str((float(phase.end_time_millis) -
+                          float(phase.start_time_millis))/1000)
+        if record.outcome == test_record.Outcome.FAIL:
+          measured = phase.measured_values
+          measurements = phase.measurements
+          for mea in measurements:
+            res = measurements[mea]
```

Comment:
`mea` and `res` are a bit nondescript. It looks like `mea` is the name (or key) of the measurement, and `res` is the actual measurement record. Maybe name these something more descriptive like `key` and `record`?